modules/cloudflarepurger/src/Plugin/Purge/DiagnosticCheck/CredentialCheck.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\purge\Plugin\PurgeDiagnosticCheck\CloudFlareCredentialCheck.
+ */
+
+namespace Drupal\cloudflarepurger\Plugin\Purge\DiagnosticCheck;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\purge\Plugin\Purge\DiagnosticCheck\DiagnosticCheckBase;
+use Drupal\purge\Plugin\Purge\DiagnosticCheck\DiagnosticCheckInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Checks if valid Api credentials have been entered for CloudFlare.
+ *
+ * @PurgeDiagnosticCheck(
+ *   id = "cloudflare_creds",
+ *   title = @Translation("CloudFlare - Credentials"),
+ *   description = @Translation("Checks to see if the credentials for CloudFlare are valid."),
+ *   dependent_queue_plugins = {},
+ *   dependent_purger_plugins = {"cloudflare"}
+ * )
+ */
+class CredentialCheck extends DiagnosticCheckBase implements DiagnosticCheckInterface {
+  /**
+   * The settings configuration.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * Constructs a CloudFlareCredentialCheck object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config
+   *   The factory for configuration objects.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->config = $config->get('cloudflare.settings');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function run() {
+    $has_valid_credentials = $this->config->get('valid_credentials');
+
+    if (!$has_valid_credentials) {
+      $this->recommendation = $this->t("Invalid Api credentials.");
+      return SELF::SEVERITY_ERROR;
+    }
+
+    $this->recommendation = $this->t('Valid Api credentials detected.');
+    return SELF::SEVERITY_OK;
```

Comment:
The code is effectively doing 80% of that.  When you save the creds via the config form it does a query against the api to test the creds.  It then saves the valid_credentials value in config.   Could potentially make it into state though